contracts/src/contracts/ACP99/ACP99Manager.sol

```diff
@@ -91,6 +86,82 @@ contract ACP99Manager is Ownable2Step, IACP99Manager {
         emit SetSecurityModule(securityModule_);
     }
 
+    /// @inheritdoc IACP99Manager
+    function initializeValidatorSet(
+        ValidatorMessages.SubnetConversionData calldata subnetConversionData,
+        uint32 messageIndex
+    ) external {
+        if (initializedValidatorSet) {
+            revert ACP99Manager__ValidatorSetAlreadyInitialized();
+        }
+        // Check that the blockchainID and validator manager address in the subnetConversionData correspond to this contract.
+        if (subnetConversionData.validatorManagerBlockchainID != warpMessenger.getBlockchainID()) {
+            revert ACP99Manager__InvalidManagerBlockchainID(
+                subnetConversionData.validatorManagerBlockchainID, warpMessenger.getBlockchainID()
+            );
+        }
+        if (address(subnetConversionData.validatorManagerAddress) != address(this)) {
+            revert ACP99Manager__InvalidManagerAddress(
+                subnetConversionData.validatorManagerAddress, address(this)
+            );
+        }
+
+        uint256 numInitialValidators = subnetConversionData.initialValidators.length;
```

Comment:
Can the message arrive with InitialValidator[] being 0, or does it always contain at least one validator?

Reply:
Good question, I believe that the P-Chain should check that the `InitialValidators` is non-empty.